### PR TITLE
[tests-only] Improve then steps for suite 'apiFederationToShares1'

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
Improve then steps for suite `apiFederationToShares1`

## Related Issue

- Part of https://github.com/owncloud/core/issues/39512

## How Has This Been Tested?

- :robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiFederationToShares1/federated.feature [hunk 1]
```diff
@@ -568,15 +570,17 @@ Feature: federated
     And user "Brian" has created folder "/zzzfolder"
     And user "Brian" has created folder "zzzfolder/local"
     And user "Brian" has uploaded file with content "local content" to "/randomfile.txt"
-    When user "Brian" shares folder "zzzfolder" with user "Carol" using the sharing API
-    And user "Carol" accepts share "/zzzfolder" offered by user "Brian" using the sharing API
+    And user "Brian" shares folder "zzzfolder" with user "Carol" using the sharing API
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiFederationToShares1/federated.feature:573]:
```suggestion
    And user "Brian" has shared folder "zzzfolder" with user "Carol" using the sharing API
```

File: tests/acceptance/features/apiFederationToShares1/federated.feature [hunk 2]
```diff
@@ -568,15 +570,17 @@ Feature: federated
...
     And user "Brian" shares folder "randomfile.txt" with user "Carol" using the sharing API
-    And user "Carol" accepts share "/randomfile.txt" offered by user "Brian" using the sharing API
     And user "Alice" from server "REMOTE" shares "zzzfolder" with user "Carol" from server "LOCAL" using the sharing API
-    And user "Carol" from server "LOCAL" accepts the last pending share using the sharing API
     And user "Alice" from server "REMOTE" shares "randomfile.txt" with user "Carol" from server "LOCAL" using the sharing API
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiFederationToShares1/federated.feature:576]: Please use the Given step for this step

File: tests/acceptance/features/apiFederationToShares1/federated.feature [hunk 3]
```diff
@@ -568,15 +570,17 @@ Feature: federated
     And user "Brian" has created folder "/zzzfolder"
     And user "Brian" has created folder "zzzfolder/local"
     And user "Brian" has uploaded file with content "local content" to "/randomfile.txt"
-    When user "Brian" shares folder "zzzfolder" with user "Carol" using the sharing API
-    And user "Carol" accepts share "/zzzfolder" offered by user "Brian" using the sharing API
+    And user "Brian" shares folder "zzzfolder" with user "Carol" using the sharing API
     And user "Brian" shares folder "randomfile.txt" with user "Carol" using the sharing API
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiFederationToShares1/federated.feature:574]:
```suggestion
    And user "Brian" has shared folder "randomfile.txt" with user "Carol" using the sharing API
```

File: tests/acceptance/features/apiFederationToShares1/federated.feature [hunk 4]
```diff
@@ -568,15 +570,17 @@ Feature: federated
...
-    And user "Carol" accepts share "/randomfile.txt" offered by user "Brian" using the sharing API
     And user "Alice" from server "REMOTE" shares "zzzfolder" with user "Carol" from server "LOCAL" using the sharing API
```
Comment (amrita-shrestha) [inline tests/acceptance/features/apiFederationToShares1/federated.feature:575]:
```suggestion
    And user "Alice" from server "REMOTE" has shared "zzzfolder" with user "Carol" from server "LOCAL" using the sharing API
```

File: tests/acceptance/features/apiFederationToShares1/federated.feature (hunk 2 above)
Comment (amrita-shrestha) [inline tests/acceptance/features/apiFederationToShares1/federated.feature:576]:
```suggestion
    And user "Alice" from server "REMOTE" has shared "randomfile.txt" with user "Carol" from server "LOCAL" using the sharing API
```

File: tests/acceptance/features/apiFederationToShares1/savePublicShare.feature [hunk 5]
```diff
@@ -104,7 +105,8 @@ Feature: Save public shares created by oC users
     And user "Alice" has added the public share created from server "REMOTE" using the sharing API
     And using server "REMOTE"
     When user "RemoteAlice" deletes public link share named "sharedlink" in file "/PARENT" using the sharing API
-    And using server "LOCAL"
+    Then the HTTP status code should be "200"
+    When using server "LOCAL"
```
Comment (SagarGi) [inline tests/acceptance/features/apiFederationToShares1/savePublicShare.feature:109]: OCS code can also be checked for sharing API

**Review by phil-davis — APPROVED** [review]

**Pull request merged**